src/code-stats-insights.ts

```diff
@@ -1,121 +1,209 @@
 import * as sourcegraph from 'sourcegraph'
-import { from, defer } from 'rxjs'
-import { switchMap, map, retry } from 'rxjs/operators'
-import { IQuery, IGraphQLResponseRoot } from './schema'
+import { from, defer, Subscription } from 'rxjs'
+import { map, retry, startWith, distinctUntilChanged, mergeAll, tap } from 'rxjs/operators'
 import gql from 'tagged-template-noop'
 import { escapeRegExp, partition, sum } from 'lodash'
+import isEqual from 'lodash/isEqual'
 import linguistLanguages from 'linguist-languages'
+import { isLinguistLanguage, parseUri, queryGraphQL } from './utils';
 
-const isLinguistLanguage = (language: string): language is keyof typeof linguistLanguages =>
-    Object.prototype.hasOwnProperty.call(linguistLanguages, language)
+/**
+ * Code stat insight settings from user/org setting cascade.
+ * */
+interface Insight {
+    title: string
+    repository?: string;
+    otherThreshold?: number
 
-const queryGraphQL = async <T = IQuery>(query: string, variables: object = {}): Promise<T> => {
-    const { data, errors }: IGraphQLResponseRoot = await sourcegraph.commands.executeCommand(
-        'queryGraphQL',
-        query,
-        variables
-    )
-    if (errors && errors.length > 0) {
-        throw new Error(errors.map(e => e.message).join('\n'))
-    }
-    return (data as any) as T
-}
-
-const parseUri = (uri: URL): { repo: string } => {
-    return { repo: uri.hostname + uri.pathname }
+    /**
+     *  Synthetic field for backward compatibility.
+     *  In first version of code stats insight this query supposed to be filled
+     *  by users in user/org settings. Now we don't need set this field because
+     *  we can derive query from repository field. But for the sake of compatibility
+     *  of existing first-version insights we left this setting.
```

Comment:
some grammar corrections
```suggestion
     *  Synthetic field for backward compatibility.
     *  In the first version of code stats insight this query was supposed to be filled
     *  by users in user/org settings. Now we don't need to set this field anymore because
     *  we can derive the query from the repository field. But for the sake of compatibility
     *  with existing first-version insights we left this setting.
```